File: project/frontend/src/types/notification.ts
// 通知类型定义
export enum NotificationType {
  ORDER_REMINDER = 'order_reminder',
  PRODUCTION_ALERT = 'production_alert',
  MATERIAL_WARNING = 'material_warning',
  EQUIPMENT_MAINTENANCE = 'equipment_maintenance',
  QUALITY_ISSUE = 'quality_issue',
  SYSTEM_NOTIFICATION = 'system_notification'
}

export enum NotificationPriority {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
  URGENT = 'urgent'
}

export enum NotificationStatus {
  PENDING = 'pending',
  SENT = 'sent',
  READ = 'read',
  FAILED = 'failed'
}

export interface Notification {
  id: number;
  title: string;
  content: string;
  notification_type: NotificationType;
  priority: NotificationPriority;
  status: NotificationStatus;
  recipient_id: number;
  sender_id?: number;
  template_id?: number;
  scheduled_at?: string;
  sent_at?: string;
  read_at?: string;
  metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface NotificationTemplate {
  id: number;
  name: string;
  title_template: string;
  content_template: string;
  notification_type: NotificationType;
  is_active: boolean;
  variables?: string[];
  created_at: string;
  updated_at: string;
}

export interface NotificationRule {
  id: number;
  name: string;
  description?: string;
  trigger_event: string;
  conditions: Record<string, any>;
  template_id: number;
  target_users?: string[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface NotificationStats {
  total_count: number;
  unread_count: number;
  sent_count: number;
  failed_count: number;
  by_type: Record<NotificationType, number>;
  by_priority: Record<NotificationPriority, number>;
}

export interface NotificationQuery {
  notification_type?: NotificationType;
  status?: NotificationStatus;
  priority?: NotificationPriority;
  start_date?: string;
  end_date?: string;
  page?: number;
  size?: number;
}

export interface NotificationCreate {
  title: string;
  content: string;
  notification_type: NotificationType;
  priority: NotificationPriority;
  recipient_id: number;
  template_id?: number;
  scheduled_at?: string;
  metadata?: Record<string, any>;
}

export interface NotificationUpdate {
  title?: string;
  content?: string;
  priority?: NotificationPriority;
  status?: NotificationStatus;
  scheduled_at?: string;
  metadata?: Record<string, any>;
}

export interface BatchNotificationCreate {
  title: string;
  content: string;
  notification_type: NotificationType;
  priority: NotificationPriority;
  recipient_ids: number[];
  template_id?: number;
  scheduled_at?: string;
  metadata?: Record<string, any>;
}

// 通知显示配置
export const NOTIFICATION_TYPE_LABELS: Record<NotificationType, string> = {
  [NotificationType.ORDER_REMINDER]: '订单提醒',
  [NotificationType.PRODUCTION_ALERT]: '生产警报',
  [NotificationType.MATERIAL_WARNING]: '物料预警',
  [NotificationType.EQUIPMENT_MAINTENANCE]: '设备维护',
  [NotificationType.QUALITY_ISSUE]: '质量问题',
  [NotificationType.SYSTEM_NOTIFICATION]: '系统通知'
};

export const NOTIFICATION_PRIORITY_LABELS: Record<NotificationPriority, string> = {
  [NotificationPriority.LOW]: '低',
  [NotificationPriority.MEDIUM]: '中',
  [NotificationPriority.HIGH]: '高',
  [NotificationPriority.URGENT]: '紧急'
};

export const NOTIFICATION_STATUS_LABELS: Record<NotificationStatus, string> = {
  [NotificationStatus.PENDING]: '待发送',
  [NotificationStatus.SENT]: '已发送',
  [NotificationStatus.READ]: '已读',
  [NotificationStatus.FAILED]: '发送失败'
};

// 通知颜色配置
export const NOTIFICATION_TYPE_COLORS: Record<NotificationType, string> = {
  [NotificationType.ORDER_REMINDER]: '#1890ff',
  [NotificationType.PRODUCTION_ALERT]: '#fa8c16',
  [NotificationType.MATERIAL_WARNING]: '#faad14',
  [NotificationType.EQUIPMENT_MAINTENANCE]: '#722ed1',
  [NotificationType.QUALITY_ISSUE]: '#f5222d',
  [NotificationType.SYSTEM_NOTIFICATION]: '#52c41a'
};

export const NOTIFICATION_PRIORITY_COLORS: Record<NotificationPriority, string> = {
  [NotificationPriority.LOW]: '#52c41a',
  [NotificationPriority.MEDIUM]: '#1890ff',
  [NotificationPriority.HIGH]: '#fa8c16',
  [NotificationPriority.URGENT]: '#f5222d'
};

export const NOTIFICATION_STATUS_COLORS: Record<NotificationStatus, string> = {
  [NotificationStatus.PENDING]: '#faad14',
  [NotificationStatus.SENT]: '#1890ff',
  [NotificationStatus.READ]: '#52c41a',
  [NotificationStatus.FAILED]: '#f5222d'
};

File: project/backend/app/schemas/notification.py
from pydantic import BaseModel, Field, validator
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum

class NotificationType(str, Enum):
    """通知类型"""
    SYSTEM = "system"
    EMAIL = "email"
    WECHAT = "wechat"
    SMS = "sms"

class NotificationPriority(str, Enum):
    """通知优先级"""
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    URGENT = "urgent"

class NotificationStatus(str, Enum):
    """通知状态"""
    PENDING = "pending"
    SENDING = "sending"
    SENT = "sent"
    FAILED = "failed"
    READ = "read"

# 通知基础模式
class NotificationBase(BaseModel):
    """通知基础模式"""
    title: str = Field(..., max_length=200, description="通知标题")
    content: str = Field(..., description="通知内容")
    notification_type: NotificationType = Field(..., description="通知类型")
    priority: NotificationPriority = Field(NotificationPriority.NORMAL, description="优先级")
    recipient_id: int = Field(..., description="接收者ID")
    recipient_email: Optional[str] = Field(None, max_length=100, description="接收者邮箱")
    recipient_phone: Optional[str] = Field(None, max_length=20, description="接收者手机号")
    recipient_wechat: Optional[str] = Field(None, max_length=100, description="接收者微信号")
    related_type: Optional[str] = Field(None, max_length=50, description="关联业务类型")
    related_id: Optional[int] = Field(None, description="关联业务ID")
    scheduled_at: Optional[datetime] = Field(None, description="计划发送时间")

class NotificationCreate(NotificationBase):
    """创建通知模式"""
    sender_id: Optional[int] = Field(None, description="发送者ID")

class NotificationUpdate(BaseModel):
    """更新通知模式"""
    title: Optional[str] = Field(None, max_length=200, description="通知标题")
    content: Optional[str] = Field(None, description="通知内容")
    status: Optional[NotificationStatus] = Field(None, description="状态")
    read_at: Optional[datetime] = Field(None, description="阅读时间")

class NotificationResponse(NotificationBase):
    """通知响应模式"""
    id: int
    status: NotificationStatus
    sender_id: Optional[int]
    created_at: datetime
    sent_at: Optional[datetime]
    read_at: Optional[datetime]
    retry_count: int
    error_message: Optional[str]
    
    class Config:
        from_attributes = True

# 通知模板模式
class NotificationTemplateBase(BaseModel):
    """通知模板基础模式"""
    name: str = Field(..., max_length=100, description="模板名称")
    title_template: str = Field(..., max_length=200, description="标题模板")
    content_template: str = Field(..., description="内容模板")
    notification_type: NotificationType = Field(..., description="通知类型")
    variables: Optional[str] = Field(None, description="模板变量说明")

class NotificationTemplateCreate(NotificationTemplateBase):
    """创建通知模板模式"""
    pass

class NotificationTemplateUpdate(BaseModel):
    """更新通知模板模式"""
    name: Optional[str] = Field(None, max_length=100, description="模板名称")
    title_template: Optional[str] = Field(None, max_length=200, description="标题模板")
    content_template: Optional[str] = Field(None, description="内容模板")
    variables: Optional[str] = Field(None, description="模板变量说明")
    is_active: Optional[bool] = Field(None, description="是否启用")

class NotificationTemplateResponse(NotificationTemplateBase):
    """通知模板响应模式"""
    id: int
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

# 通知规则模式
class NotificationRuleBase(BaseModel):
    """通知规则基础模式"""
    name: str = Field(..., max_length=100, description="规则名称")
    description: Optional[str] = Field(None, description="规则描述")
    trigger_event: str = Field(..., max_length=100, description="触发事件")
    trigger_conditions: Optional[str] = Field(None, description="触发条件")
    template_id: int = Field(..., description="模板ID")
    notification_types: str = Field(..., max_length=200, description="通知类型列表")
    priority: NotificationPriority = Field(NotificationPriority.NORMAL, description="优先级")
    recipient_rules: str = Field(..., description="接收者规则")

class NotificationRuleCreate(NotificationRuleBase):
    """创建通知规则模式"""
    pass

class NotificationRuleUpdate(BaseModel):
    """更新通知规则模式"""
    name: Optional[str] = Field(None, max_length=100, description="规则名称")
    description: Optional[str] = Field(None, description="规则描述")
    trigger_conditions: Optional[str] = Field(None, description="触发条件")
    notification_types: Optional[str] = Field(None, max_length=200, description="通知类型列表")
    priority: Optional[NotificationPriority] = Field(None, description="优先级")
    recipient_rules: Optional[str] = Field(None, description="接收者规则")
    is_active: Optional[bool] = Field(None, description="是否启用")

class NotificationRuleResponse(NotificationRuleBase):
    """通知规则响应模式"""
    id: int
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

# 批量发送模式
class BatchNotificationCreate(BaseModel):
    """批量创建通知模式"""
    template_id: int = Field(..., description="模板ID")
    notification_types: List[NotificationType] = Field(..., description="通知类型列表")
    priority: NotificationPriority = Field(NotificationPriority.NORMAL, description="优先级")
    recipients: List[int] = Field(..., description="接收者ID列表")
    template_data: Dict[str, Any] = Field({}, description="模板数据")
    related_type: Optional[str] = Field(None, description="关联业务类型")
    related_id: Optional[int] = Field(None, description="关联业务ID")
    scheduled_at: Optional[datetime] = Field(None, description="计划发送时间")

# 通知统计模式
class NotificationStats(BaseModel):
    """通知统计模式"""
    total_count: int = Field(..., description="总数量")
    pending_count: int = Field(..., description="待发送数量")
    sent_count: int = Field(..., description="已发送数量")
    failed_count: int = Field(..., description="发送失败数量")
    read_count: int = Field(..., description="已读数量")
    unread_count: int = Field(..., description="未读数量")

# 通知查询模式
class NotificationQuery(BaseModel):
    """通知查询模式"""
    notification_type: Optional[NotificationType] = Field(None, description="通知类型")
    status: Optional[NotificationStatus] = Field(None, description="状态")
    priority: Optional[NotificationPriority] = Field(None, description="优先级")
    recipient_id: Optional[int] = Field(None, description="接收者ID")
    sender_id: Optional[int] = Field(None, description="发送者ID")
    related_type: Optional[str] = Field(None, description="关联业务类型")
    related_id: Optional[int] = Field(None, description="关联业务ID")
    start_date: Optional[datetime] = Field(None, description="开始日期")
    end_date: Optional[datetime] = Field(None, description="结束日期")
    page: int = Field(1, ge=1, description="页码")
    size: int = Field(20, ge=1, le=100, description="每页数量")